example/lib/pages/popup_page.dart

```diff
@@ -37,32 +39,48 @@ class _PopupPageState extends State<PopupPage> {
               ];
             },
           ),
-          const SizedBox(
-            width: 10,
+          YaruPopupMenuButton<MyEnum>(
+            onSelected: (value) {
+              if (enumSet.contains(value)) {
+                enumSet.remove(value);
+              } else {
+                enumSet.add(value);
+              }
+            },
+            child: const Text('Multi Select'),
+            itemBuilder: (context) {
+              return [
+                for (final value in MyEnum.values)
+                  YaruCheckedPopupMenuItem<MyEnum>(
+                    padding: EdgeInsets.zero,
+                    value: value,
+                    checked: enumSet.contains(value),
+                    child: Text(value.name),
+                  )
+              ];
+            },
           ),
-          StatefulBuilder(
-            builder: (context, setState) {
-              return YaruPopupMenuButton<MyEnum>(
-                onSelected: (value) {
-                  if (enumSet.contains(value)) {
-                    enumSet.remove(value);
-                  } else {
-                    enumSet.add(value);
-                  }
-                },
-                child: const Text('Multi Select'),
-                itemBuilder: (context) {
-                  return [
-                    for (final value in MyEnum.values)
-                      YaruCheckedPopupMenuItem<MyEnum>(
-                        padding: EdgeInsets.zero,
-                        value: value,
-                        checked: enumSet.contains(value),
-                        child: Text(value.name),
-                      )
-                  ];
-                },
-              );
+          YaruPopupMenuButton<MyEnum>(
+            onSelected: (value) {
+              if (enumSet.contains(value)) {
+                enumSet.remove(value);
+              } else {
+                enumSet.add(value);
+              }
```

Comment:
unfortunately, `onSelected` is meaningless in multi-selection mode. should this code be in `onChanged` to keep `enumSet` in sync and?

```
onChanged: (checked) {
  if (checked) {
    enumSet.add(value);
  } else {
    enumSet.remove(value);
  }
},
```

Reply:
woops yes that slipped through

Just thought that it would be cool to insert code snippets to the yaru widgets example :P Do you know how to force dart syntax in flutter_markdown? ```dart does not work

Reply:
Yes, that would be awesome! Hmm, check the Flutter gallery - I recall it has some example snippets

Reply:
ok I take this out of here for now. would be cool to find a generic approach for all pages then